samcli/commands/local/get_env_vars/cli.py

```diff
@@ -0,0 +1,68 @@
+"""
+CLI command for "get-env-vars" command
+"""
+
+import copy
+import json
+import logging
+
+import boto3
+import click
+
+from samcli.cli.main import pass_context, common_options as cli_framework_options, aws_creds_options
+
+LOG = logging.getLogger(__name__)
+
+HELP_TEXT = """
+You can use this command to generate a JSON file of environment vars for
+each Lambda function in an existing CloudFormation stack in the format expected
+by the --env-vars option of the local invoke commands. This is useful for locally
+testing Lambda function code against dependencies that are deployed in the cloud.\n
+\b
+Generating an environment variable JSON file from stack my-stack
+$ sam local get-env-vars --stack-name my-stack > env.json\n
+"""
+
+
+@click.command("get-env-vars", help=HELP_TEXT, short_help="Generate environment variables JSON from a stack")
+@click.option("--stack-name",
+              required=True,
+              help="Name or ARN of the AWS CloudFormation stack to use to generate the environment variables JSON.")
+@cli_framework_options
+@aws_creds_options
+@pass_context
+def cli(ctx,
+        stack_name
+        ):
+    # All logic must be implemented in the ``do_cli`` method. This helps with easy unit testing
+
+    do_cli(stack_name)  # pragma: no cover
+
+
+def do_cli(stack_name):
+    """
+    Implementation of the ``cli`` method
+    """
+
+    LOG.debug("'get-env-vars' command is called")
+
+    cfn_client = boto3.client('cloudformation')
+    lambda_client = boto3.client('lambda')
+
+    stack_resource_iterator = cfn_client.get_paginator('list_stack_resources').paginate(
+        StackName=stack_name
+    )
+
+    function_resource_summaries = [summary for response in stack_resource_iterator
```

Comment:
do we need to consider AWS::Serverless::Function?

Reply:
No. This is a query on an actual deployed CloudFormation stack. `AWS::Serverless::Function` is a pseudo-resource of SAM, but the actual deployed stack will only contain `AWS::Lambda::Function` resources.